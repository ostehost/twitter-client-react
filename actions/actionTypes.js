const actionTypes = {
    CHANGE_SEARCH: 'CHANGE_SEARCH',
    CHANGE_COUNT: 'CHANGE_COUNT',
    CHANGE_TYPE: 'CHANGE_TYPE',
    LOAD_TWEETS: 'LOAD_TWEETS',
    TWEETS_LOADED: 'TWEETS_LOADED',
    LOAD_TWEETS_FAILURE: 'LOAD_TWEETS_FAILURE',
    SORT: 'SORT'
}

export default actionTypes;