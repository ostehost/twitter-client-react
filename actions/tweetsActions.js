import actionTypes from './actionTypes';

export const searchChange = (value) => (
    { type: actionTypes.CHANGE_SEARCH, value: value }
)

export const typeChange = (value) => (
    { type: actionTypes.CHANGE_TYPE, value: value }
)

export const countChange = (value) => (
    { type: actionTypes.CHANGE_COUNT, value: value }
)

export const loadTweets = () => (
    { type: actionTypes.LOAD_TWEETS }
)

export const tweetsLoaded = (payload, append) => (
    { type: actionTypes.TWEETS_LOADED, payload, append }
)

export const sort = (sortBy, sortOrder) => (
    { type: actionTypes.SORT, sortBy, sortOrder }
)

export const loadTweetsFailure = (error) => {
    return { type: actionTypes.LOAD_TWEETS_FAILURE, error }
}