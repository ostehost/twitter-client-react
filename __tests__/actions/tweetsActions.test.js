import configureStore from 'redux-mock-store';

// Actions to be tested
import * as tweetsActions from '../../actions/tweetsActions';
import actionTypes from '../../actions/actionTypes';

const mockStore = configureStore();
const store = mockStore();


describe('tweetsActions', () => {
    beforeEach(() => { // Runs before each test in the suite
        store.clearActions();
    });

    describe('searchChange', () => {
        test('Dispatches the correct action and payload', () => {
            store.dispatch(tweetsActions.searchChange("value"));
            expect(store.getActions()).toMatchSnapshot();

            // const expectedActions = [
            //     {
            //         'value': "value",
            //         'type': actionTypes.CHANGE_SEARCH,
            //     },
            // ];

            // store.dispatch(tweetsActions.searchChange("value"));
            // expect(store.getActions()).toEqual(expectedActions);
        });
    });

    describe('typeChange', () => {
        test('Dispatches the correct action and payload', () => {
            store.dispatch(tweetsActions.typeChange("value"));
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    describe('countChange', () => {
        test('Dispatches the correct action and payload', () => {
            store.dispatch(tweetsActions.countChange("value"));
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    describe('loadTweets', () => {
        test('Dispatches the correct action and payload', () => {
            store.dispatch(tweetsActions.loadTweets());
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    describe('tweetsLoaded', () => {
        test('Dispatches the correct action and payload', () => {
            store.dispatch(tweetsActions.tweetsLoaded("value", true));
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    describe('sort', () => {
        test('Dispatches the correct action and payload', () => {
            store.dispatch(tweetsActions.sort('created_at', -1));
            expect(store.getActions()).toMatchSnapshot();
        });
    });

    describe('loadTweetsFailure', () => {
        test('Dispatches the correct action and payload', () => {
            store.dispatch(tweetsActions.loadTweetsFailure('error'));
            expect(store.getActions()).toMatchSnapshot();
        });
    });
});