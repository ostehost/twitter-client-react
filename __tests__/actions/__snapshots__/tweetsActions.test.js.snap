// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tweetsActions countChange Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "CHANGE_COUNT",
    "value": "value",
  },
]
`;

exports[`tweetsActions loadTweets Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "LOAD_TWEETS",
  },
]
`;

exports[`tweetsActions loadTweetsFailure Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "error": "error",
    "type": "LOAD_TWEETS_FAILURE",
  },
]
`;

exports[`tweetsActions searchChange Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "CHANGE_SEARCH",
    "value": "value",
  },
]
`;

exports[`tweetsActions sort Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "sortBy": "created_at",
    "sortOrder": -1,
    "type": "SORT",
  },
]
`;

exports[`tweetsActions tweetsLoaded Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "append": true,
    "payload": "value",
    "type": "TWEETS_LOADED",
  },
]
`;

exports[`tweetsActions typeChange Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "type": "CHANGE_TYPE",
    "value": "value",
  },
]
`;
